auth/oauth.go

```diff
@@ -40,6 +40,21 @@ type OAuthFlow struct {
 	TimeSleep        func(time.Duration)
 }
 
+func detectDeviceFlow(statusCode int, values url.Values) (bool, error) {
```

Comment:
What are your thoughts on naming this `tryFallbackAuth` or `fallbackAuthRequired`?

Reply:
I don't feel those names are as accurate, since this method neither "tries" to perform fallback auth nor detects whether fallback auth is "required". It detects whether Device Flow is supported on the remote, errors out on legitimate HTTP errors, and will fall back in the rest of the cases.